src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -456,7 +454,7 @@ private int GetServicePortValue(V1ServicePort servicePort, V1Container container
                             DnsName = isInWorkloadNamespace ?
                                         $"{address.Hostname}.{endpoint.Metadata.Name}" :
                                         $"{address.Hostname}.{endpoint.Metadata.Name}.{endpoint.Metadata.Namespace()}",
-                            Ports = ports,
+                            Ports = subset.Ports?.Where(port => this._IsSupportedProtocol(port.Protocol, endpoint.Metadata.Name)).Select(p => new PortPair(remotePort: p.Port)).ToArray() ?? new PortPair[] { },

```

Comment:
Hi Elena, thanks for including me. so now instead of setting ports for each subset of endpoints, you have moved it to for each of list of address for endpoints ? I see it is same statement moved from the top, wondering if this worked ?

Reply:
so the original assignment was correct in terms of content, only problem was because it was same object it shared same memory address (this is a C# thing of shallow copy vs deep copy all object except structs in c# are by reference). By moving it here now content is still original content it was using before but each endpoint gets its own object (new keyword for each assignment) so later on changing the value of a port on any endpoint wont change the value of the port in all endpoints.